rpctest/utils.go

```diff
@@ -76,24 +77,25 @@ func syncMempools(nodes []*Harness) error {
 	return nil
 }
 
-// syncBlocks blocks until all nodes report the same block height.
+// syncBlocks blocks until all nodes report the same best chain.
 func syncBlocks(nodes []*Harness) error {
 	blocksMatch := false
 
+retry:
 	for !blocksMatch {
-	retry:
-		blockHeights := make(map[int64]struct{})
+		blocksByHash := make(map[chainhash.Hash]int32)
 
 		for _, node := range nodes {
-			blockHeight, err := node.Node.GetBlockCount()
+			blockHash, blockHeight, err := node.Node.GetBestBlock()
 			if err != nil {
 				return err
 			}
 
-			blockHeights[blockHeight] = struct{}{}
-			if len(blockHeights) > 1 {
+			existingHeight, ok := blocksByHash[*blockHash]
```

Comment:
Actually, after thinking about it (and seeing the test fail upon changing it), it is required.  The reason is because you only want to fail the check when another node actually has the given block hash as at given height that is not the current node's `blockHeight`.

If it's removed, the test fails like so:
```
--- FAIL: TestHarness (61.71s)
	rpc_harness_test.go:324: blocks never detected as synced
```